### PR TITLE
MP photogram industrial 2

### PR DESCRIPTION
**Pull request by Ritumka**

### CODE REVIEW

File: app/controllers/photos_controller.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    # Use callbacks to share common setup or constraints between actions.
+    def set_photo
+      @photo = Photo.find(params[:id])
+    end
+
+    # Only allow a list of trusted parameters through.
+    def photo_params
+      params.require(:photo).permit(:image, :comments_count, :likes_count, :caption, :owner_id)
+    end
+end
```
Comment (JoseMDO) [inline app/controllers/photos_controller.rb]: All of your controllers look great Margo!

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+#
+#  index_comments_on_photo_id  (photo_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (author_id => users.id)
+#  fk_rails_...  (photo_id => photos.id)
+#
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: "User", counter_cache: true
```
Comment (JoseMDO) [inline app/models/comment.rb:22]: Associations are good

File: app/models/follow_request.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+#  index_follow_requests_on_recipient_id  (recipient_id)
+#  index_follow_requests_on_sender_id     (sender_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (recipient_id => users.id)
+#  fk_rails_...  (sender_id => users.id)
+#
+class FollowRequest < ApplicationRecord
+  belongs_to :recipient, class_name: "User"
```
Comment (JoseMDO) [inline app/models/follow_request.rb:23]: 👍

File: app/models/photo.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+#
+# Foreign Keys
+#
+#  fk_rails_...  (owner_id => users.id)
+#
+class Photo < ApplicationRecord
+  belongs_to :owner, class_name: "User", counter_cache: true
+  has_many :comments
+  has_many :likes
+  has_many :fans, through: :likes
```
Comment (JoseMDO) [inline app/models/photo.rb:26]: Nice

File: app/models/user.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+  has_many :liked_photos, through: :likes, source: :photo
+
+  has_many :leaders, through: :accepted_sent_follow_requests, source: :recipient
+
+  has_many :followers, through: :accepted_received_follow_requests, source: :sender
+
+  has_many :feed, through: :leaders, source: :own_photos
+
+  has_many :discover, through: :leaders, source: :liked_photos
```
Comment (JoseMDO) [inline app/models/user.rb:48]: Nice job!

File: config/routes.rb [hunk 6]
```diff
@@ -1,4 +1,9 @@
 Rails.application.routes.draw do
+  resources :likes
```
Comment (JoseMDO) [inline config/routes.rb:2]: Routes look great!

**Review by JoseMDO — APPROVED** [review]
Awesome job Margo!

**Review by merrillhuang — APPROVED** [review]
Everything looks good!

**Pull request merged**